### PR TITLE
SIMD: SSE2 intrinsic implementation for float64 input of np.einsum

### PR DESCRIPTION
**Pull request by Qiyu8**
`np.enisum` is a powerful function like the Swiss Army Knife that implements many different functions, But two defects were found:

- 10 sub modules of `np.enisum`  has been optimized by using SSE1, but only 6 of them has the corresponding SSE2 implementation.
- No benchmark test cases was found to measure the extent of performance improvement of SIMD implementation.

This PR turns to solve above problems:

- Add 4 missing SSE2 implementation. A total of 10+% performance increased.
- Add several benchmark test cases to measure SIMD performance.

Here is the benchmark result.
<details>

Before optimize(master branch):

```
[ 81.25%] ··· bench_linalg.Einsum.time_einsum_multiply                                                                                                                                         ok
[ 81.25%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   85.4±8μs
              =============== ==========

[ 87.50%] ··· bench_linalg.Einsum.time_einsum_outer                                                                                                                                            ok 
[ 87.50%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   53.0±4ms
              =============== ==========

[ 93.75%] ··· bench_linalg.Einsum.time_einsum_sum_mul                                                                                                                                          ok 
[ 93.75%] ··· =============== ===========
                   dtype
              --------------- -----------
               numpy.float64   86.8±20μs
              =============== ===========

[100.00%] ··· bench_linalg.Einsum.time_einsum_sum_mul2                                                                                                                                         ok 
[100.00%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   68.8±9μs
              =============== ==========
```
After Optimize(enisum_sse2 branch):
```
[ 56.25%] ··· bench_linalg.Einsum.time_einsum_multiply                                                                                                                                         ok
[ 56.25%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   72.3±8μs
              =============== ==========

[ 62.50%] ··· bench_linalg.Einsum.time_einsum_outer                                                                                                                                            ok
[ 62.50%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   47.8±2ms
              =============== ==========

[ 68.75%] ··· bench_linalg.Einsum.time_einsum_sum_mul                                                                                                                                          ok
[ 68.75%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   65.8±3μs
              =============== ==========

[ 75.00%] ··· bench_linalg.Einsum.time_einsum_sum_mul2                                                                                                                                         ok
[ 75.00%] ··· =============== ==========
                   dtype
              --------------- ----------
               numpy.float64   62.4±6μs
              =============== ==========
```

### CODE REVIEW

**Comment by charris** [comment]
Not sure why aarch64 failed the first time around and passed the second time. Hmm...

**Comment by Qiyu8** [comment]
> Not sure why aarch64 failed the first time around and passed the second time. Hmm...

That's strange, because the code that changes here only affects the X86 platform.

**Comment by charris** [comment]
> changes here only affects the X86 platform

I'm suspecting OpenBLAS, but it could be something else.

**Review by anirudh2290 — APPROVED** [review]
I see a 2 to 8 percent improvement on your benchmarks. Also, the PR changes LGTM !

File: benchmarks/benchmarks/bench_linalg.py [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 import numpy as np
 
-
```
Comment (mattip) [inline benchmarks/benchmarks/bench_linalg.py:5]: [PEP 8](https://www.python.org/dev/peps/pep-0008/#blank-lines) recommends two blank lines before top-level classes and functions.

**Comment by mattip** [comment]
Thanks @Qiyu8

**Pull request merged**